src/adapter_test_helpers/sumo_basic_test_helper.erl

```diff
@@ -176,6 +177,24 @@ check_proper_dates(Config) ->
   Date = Module:birthdate(Person),
   ok.
 
+-spec persist_using_changeset(config()) -> ok.
+persist_using_changeset(Config) ->
+  {_, Name} = lists:keyfind(name, 1, Config),
+  Module = sumo_config:get_prop_value(Name, module),
+  Schema = Module:sumo_schema(),
+
+  [] = sumo:find_by(Name, [{name, <<"John">>}]),
+  [P1] = sumo:find_by(Name, [{name, <<"A">>}]),
+  Doc1 = sumo:new_doc(people, P1),
```

Comment:
Why do you require users to generate `sumo_internal:doc` representations of their objects to use `sumo_changeset:cast`?
Personally, I would prefer

```erlang
    sumo_changeset:cast(people, P1, #{name => <<"John">>}, Allowed),
```
…over…
```erlang
    Doc1 = sumo:new_doc(people, P1),
    CS1 = sumo_changeset:cast(Doc1, #{name => <<"John">>}, Allowed),
```